scripts/custodian/README.md

```diff
@@ -5,18 +5,64 @@ QA custodian will clean up resources in AWS, GCR, and AZURE. Currently, we have
+
+We tie these keys into the cloud custodian with a simple `sed` command in the dockerfile that replaces each of the above keys in the yaml files with the values of said keys. This decision was made to keep any sensitive data separate from the code. 
+
+
+### Important Values In The Code
+* [Dockerfile](./Dockerfile)
+  * a `sed` command replaces the hardcoded key names with the values from said key. These are arguments sent to the dockerfile at build time
+  * (AWS explicit) hardcoded regions are ones we use on a regular basis, and therefore what the custodian will check against
+  * for google cloud, `google_credentials.json` is a file local to where you run docker build. Also, the google cloud project is hard coded to our QA project.
+  * `debugging`, add the flag `--dry-run` to each line in the dockerfile that starts with `custodian run`. Otherwise, running the container will modify resources in your cloud!! Debug/dry-run mode will report the number of resources which would have been modified
+* `.yaml` files are built around the instructions from the [How It Works](#how-it-works) documentation from cloud custodian, which often adds new support for time-based management of resources. 
+* [Jenkinsfile](./Jenkinsfile), specifically the first docker build line, to see which variables are necessary to build the container. 
+  * **Note** for google cloud, `google_credentials.json` is a file local to where you run docker build. Also, the google cloud project is hard coded to our QA project.
+
+
+
+### Non-Implemented Features
+
+#### Tag-To-Save
+tag-to-save is a yaml that is setup to block resources from being deleted untl untag-to-delete runs. Management of runtime would be through jenkins:
+* untag job runs at EOD friday to cleanup any resources that were tagged to be saved earlier in the week (no resouces should be saved over the weekend)
+* tagged job would be ran manually, where user(s) would specify their key or resource(s) which would be tagged to be saved until Friday
+however, no one has had any request to save anything beyond 2 days (or at least not voiced it in any channel) so this feature isn't implemented to save on complexity/potential issues until it is needed
+
+#### Linode
+
+Originally, this was a hack-week project where linode resource management was done via its cli (Cloud Custodian doesn't support linode) however I ripped this out except for one part, the build args to the dockerfile. Doing resource management through the CLI would be difficult unless we used an api-key with org-level access to resources. Currently for QA, linode is the only cloud we use that is user-scoped correctly, meaning that my key doesn't have access to see other users resources in our org. Generally a good thing, but not for cleanup. So this is left as a placeholder until we want to manage linode resources too. This would be a separate effort, unless Cloud Custodian decides to add support for it. 
```

Comment:
> however I ripped this out... [...] ...meaning that my key doesn't have access

tiniest super-nit: once this is merged, it will be unclear who the first person in this case is, so may want to phrase differently.

On this one, we could use an admin key -- potentially even a user scoped for deletion only? I don't know if this is possible, but I added it to my TODOs to look into.